src/accountsdb/download.zig

```diff
@@ -295,81 +297,91 @@ pub fn downloadSnapshotsFromGossip(
 }
 
 const DownloadProgress = struct {
-    mmap: []align(std.mem.page_size) u8,
-    download_size: usize,
+    file: std.fs.File,
     min_mb_per_second: ?usize,
     logger: Logger,
 
     mb_timer: std.time.Timer,
-    bytes_read: usize = 0,
-    file_memory_index: usize = 0,
+    bytes_read: u64 = 0,
+    total_read: u64 = 0,
     has_checked_speed: bool = false,
 
     const Self = @This();
 
-    pub fn init(
+    fn init(
         logger: Logger,
         output_dir: std.fs.Dir,
         filename: []const u8,
         download_size: usize,
         min_mb_per_second: ?usize,
     ) !Self {
-        const file = try output_dir.createFile(filename, .{ .read = true });
-        defer file.close();
+        const file = try output_dir.createFile(filename, .{});
 
         // resize the file
         try file.seekTo(download_size - 1);
         _ = try file.write(&[_]u8{1});
         try file.seekTo(0);
 
-        const file_memory = try std.posix.mmap(
```

Comment:
Just curious, do you know specifically why mmap doesn't work as well as directly writing to the file? Also, have you measured any performance differences?

Reply:
Here is my reasoning
1. There will be virtually no performance difference between using `write` and `mmap`. The difference between the two is when the kernel potentially blocks for flush / disk-io. `write` makes this explicit, while `mmap` makes this implicit. When the flush itself happens depends on the kernel implementation details of the page cache.
2. Given that ^, there is no downside to use `write`, and I believe the code is simpler to understand and more compact. We don't need to worry about passing the right flags to `mmap` etc.